internal/datastore/memdb/memdb.go

```diff
@@ -105,11 +105,11 @@ func (mdb *memdbDatastore) SnapshotReader(dr datastore.Revision) datastore.Reade
 	defer mdb.RUnlock()
 
 	if len(mdb.revisions) == 0 {
-		return &memdbReader{nil, nil, fmt.Errorf("memdb datastore is not ready")}
+		return &memdbReader{nil, nil, fmt.Errorf("memdb datastore is not ready"), time.Now()}
```

Comment:
we could accidentally instantiate `memdbReader` without transaction timestamp, and impact expiration logic. Please create constructor function that forces us to add it.

Reply:
I checked: its forced by virtue of us using the structural form everywhere and this is internal to the package